### DISCUSSION TITLE
propagating error from scoreboard to test fail

### DISCUSSION THREAD

**Discussion by Raghupal123** [raysalemi/Python4RTLVerification#3] [Q&A]
Hi Ray,

Have many questions on the pyuvm, want to start with this. In one of test scoreborad failed but test is passing, what is the recommended way to propagate this fail to the testlevel?

180.00ns ERROR    my_scoreboard.py(32)[uvm_test_top.env.scoreboard]:  FAILED: actual_result = 0x253efb ref_result = 0x300253efb
   180.00ns INFO     my_test passed
   180.00ns INFO     **************************************************************************************
                     ** TEST                          STATUS  SIM TIME (ns)  REAL TIME (s)  RATIO (ns/s) **
                     **************************************************************************************
                     ** my_test.my_test                PASS         180.00           0.02       8391.17  **
                     **************************************************************************************
                     ** TESTS=1 PASS=1 FAIL=0 SKIP=0                180.00           0.80        224.15  **
                     **************************************************************************************

**Comment by raysalemi** [comment]
> Hi Ray,
> 
> 
> 
> Have many questions on the pyuvm, want to start with this. In one of test scoreborad failed but test is passing, what is the recommended way to propagate this fail to the testlevel?
> 
> 
> 
> 180.00ns ERROR    my_scoreboard.py(32)[uvm_test_top.env.scoreboard]:  FAILED: actual_result = 0x253efb ref_result = 0x300253efb
> 
>    180.00ns INFO     my_test passed
> 
>    180.00ns INFO     **************************************************************************************
> 
>                      ** TEST                          STATUS  SIM TIME (ns)  REAL TIME (s)  RATIO (ns/s) **
> 
>                      **************************************************************************************
> 
>                      ** my_test.my_test                PASS         180.00           0.02       8391.17  **
> 
>                      **************************************************************************************
> 
>                      ** TESTS=1 PASS=1 FAIL=0 SKIP=0                180.00           0.80        224.15  **
> 
>                      **************************************************************************************
> 
> 

Cocotb uses failed assert statement to propagate status. 

Try adding `assert false` to your failure code.

**Reply by Raghupal123** [reply]
That did work, thank you
